src/platform/packages/shared/shared-ux/toolbar_selector/src/toolbar_selector.test.tsx

```diff
@@ -0,0 +1,139 @@
+        singleSelection={true}
+      />
+    );
+    fireEvent.click(getByTestId('toolbarSelectorTestButton'));
+    expect(getByText('Option 1')).toBeInTheDocument();
+    expect(getByText('Option 2')).toBeInTheDocument();
+    expect(getByText('Option 3')).toBeInTheDocument();
+  });
+
+  it('calls onChange for single selection', () => {
+    const onChange = jest.fn();
+    const { getByTestId, getByText } = render(
+      <ToolbarSelector
+        data-test-subj="toolbarSelectorTest"
+        buttonLabel="Select"
+        options={options}
+        searchable={false}
+        singleSelection={true}
+        onChange={onChange}
+      />
+    );
+    fireEvent.click(getByTestId('toolbarSelectorTestButton'));
+    fireEvent.click(getByText('Option 2'));
+    expect(onChange).toHaveBeenCalled();
+    expect(onChange.mock.calls[0][0].label).toBe('Option 2');
+  });
+
+  it('calls onChange for multi selection', () => {
+    const onChange = jest.fn();
+    const multiOptions: SelectableEntry[] = [
+      { label: 'A', value: 'a' },
+      { label: 'B', value: 'b' },
+    ];
+    const { getByTestId, getByText } = render(
+      <ToolbarSelector
+        data-test-subj="toolbarSelectorMultiTest"
+        buttonLabel="Multi"
+        options={multiOptions}
+        searchable={false}
+        singleSelection={false}
+        onChange={onChange}
+      />
+    );
+    fireEvent.click(getByTestId('toolbarSelectorMultiTestButton'));
+    fireEvent.click(getByText('A'));
+    fireEvent.click(getByText('B'));
+    expect(onChange).toHaveBeenCalled();
+    // Should be called with array of selected options
+    expect(Array.isArray(onChange.mock.calls[0][0])).toBe(true);
+  });
+
+  it('renders with disabled state', () => {
+    const { getByTestId } = render(
+      <ToolbarSelector
+        data-test-subj="toolbarSelectorDisabled"
+        buttonLabel="Disabled"
+        options={options}
+        searchable={false}
+        singleSelection={true}
+        disabled={true}
+      />
+    );
+    expect(getByTestId('toolbarSelectorDisabledButton')).toBeDisabled();
+  });
+
+  it('renders with searchable enabled', () => {
+    const { getByTestId } = render(
+      <ToolbarSelector
+        data-test-subj="toolbarSelectorSearch"
+        buttonLabel="Search"
+        options={options}
+        searchable={true}
+        singleSelection={true}
+      />
+    );
+    fireEvent.click(getByTestId('toolbarSelectorSearchButton'));
+    expect(getByTestId('toolbarSelectorSearchSelectorSearch')).toBeInTheDocument();
+  });
+
+  it('renders with optional props: popoverTitle, clearAllSection, hasArrow', () => {
+    const { getByTestId, getAllByLabelText, getByText } = render(
+      <ToolbarSelector
+        data-test-subj="toolbarSelectorOptionalProps"
+        buttonLabel="Optional"
+        options={options}
+        searchable={true}
+        singleSelection={true}
+        popoverTitle="My Popover Title"
+        clearAllSection={<span>Custom Section</span>}
```

Comment:
If `clearAllSection` accepts any component - not necessarily a component that has a selection counter and a clear all button -, should we rename the prop to something more generic but clear enough to tell it will be rendered below the search bar?

Reply:
maybe `searchProps: { append: React.ReactNode } }`?

Reply:
We already have search props > append in a different context when using search 🤔 And it's not directly inside the search, so I think `popoverContentBelowSearch` describes it well. Or maybe just `contentBelowSearch`, idk